PsfShimMonitor/KernelTrace.cs

```diff
@@ -0,0 +1,1356 @@
+#if DEBUG
+                                                 if (_ModelEventItems.Count < 10000)
+#else
+                                                 if (FilterOnProcessId == pid)
+#endif
+                                                 {
+                                                     try
+                                                     {
+                                                         string inputs = "DiskNumber=    \t0x" + ((int)data.PayloadByName("DiskNumber")).ToString("x");
+                                                         inputs += "\nIrpFlags=    \t" + Interpret_KernelFile_IrpFlags((Microsoft.Diagnostics.Tracing.Parsers.Kernel.IrpFlags)data.PayloadByName("IrpFlags"));
+                                                         inputs += "\nPriority=    \t" + Interpret_KernelFile_Priority((Microsoft.Diagnostics.Tracing.Parsers.Kernel.IOPriority)data.PayloadByName("Priority"));
+                                                         inputs += "\nTransferSize= \t0x" + ((int)data.PayloadByName("Priority")).ToString("x");
+                                                         inputs += "\nByteOffset=   \t0x" + ((long)data.PayloadByName("ByteOffset")).ToString("x");
+                                                         inputs += "\nIrp=      \t0x" + ((ulong)data.PayloadByName("Irp")).ToString("x");
+                                                         inputs += "\nFileKey=    \t0x" + ((ulong)data.PayloadByName("FileKey")).ToString("x");
+
+                                                         string outputs = "ElapsedTimeMS=\t" + data.PayloadStringByName("ElapsedTimeMS");
+                                                         outputs += "\nDiskServiceTimeMS=\t" + data.PayloadStringByName("DiskServiceTimeMSec");
+
+
+                                                         EventItem ei = new EventItem(data, inputs, "", outputs, "");
+                                                         lock (_TKernelEventListsLock)
+                                                         {
+                                                             _TKernelEventListItems.Add(ei);
+                                                         }
+                                                         worker.ReportProgress((int)data.EventIndex);
+                                                     }
+                                                     catch (Exception ex)
+                                                     {
+                                                         ;
+                                                     }
+                                                 }
+                                             }
+                                             else if (data.EventName.StartsWith("DiskIO/FlushInit"))
+                                             {
+  
```

Comment:
The error you assign to s does not equal the comment for case 3.  Is the error ERROR_PATH_NOT_FOUND or ERROR_FILE_NOT_FOUND?

Reply:
done